File: BackEnd/backend/src/main/java/com/fitapp/backend/category/aplication/dto/request/ExerciseCategoryFilterRequest.java
package com.fitapp.backend.category.aplication.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import org.springframework.data.domain.Sort;

@Data
@Schema(description = "Filtros para búsqueda de categorías")
public class ExerciseCategoryFilterRequest {
    
    @Schema(description = "Texto de búsqueda")
    @JsonProperty("search")
    private String search;
    
    @Schema(description = "Si es predefinida")
    @JsonProperty("isPredefined")
    private Boolean isPredefined;
    
    @Schema(description = "Si está activa")
    @JsonProperty("isActive")
    private Boolean isActive;
    
    @Schema(description = "Si es pública")
    @JsonProperty("isPublic")
    private Boolean isPublic;
    
    @Schema(description = "ID del deporte")
    @JsonProperty("sportId")
    private Long sportId;
    
    @Schema(description = "ID del dueño")
    @JsonProperty("ownerId")
    private Long ownerId;
    
    @Schema(description = "Solo mis categorías")
    @JsonProperty("onlyMine")
    private Boolean onlyMine = false;
    
    @Schema(description = "Incluir predefinidas")
    @JsonProperty("includePredefined")
    private Boolean includePredefined = true;
    
    @Schema(description = "Página solicitada", example = "0")
    @JsonProperty("page")
    private Integer page = 0;
    
    @Schema(description = "Tamaño de página", example = "20")
    @JsonProperty("size")
    private Integer size = 20;
    
    @Schema(description = "Campo para ordenar", example = "name")
    @JsonProperty("sortBy")
    private String sortBy = "name";
    
    @Schema(description = "Dirección del orden", example = "ASC")
    @JsonProperty("direction")
    private Sort.Direction direction = Sort.Direction.ASC;
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/response/category/request/ExerciseCategoryFilterRequest.kt
package com.fitapp.appfit.response.category.request

import com.google.gson.annotations.SerializedName

data class ExerciseCategoryFilterRequest(
    @SerializedName("search") val search: String? = null,
    @SerializedName("isPredefined") val isPredefined: Boolean? = null,
    @SerializedName("isActive") val isActive: Boolean? = null,
    @SerializedName("isPublic") val isPublic: Boolean? = null,
    @SerializedName("sportId") val sportId: Long? = null,
    @SerializedName("ownerId") val ownerId: Long? = null,
    @SerializedName("onlyMine") val onlyMine: Boolean = false,
    @SerializedName("includePredefined") val includePredefined: Boolean = true,
    @SerializedName("page") val page: Int = 0,
    @SerializedName("size") val size: Int = 20,
    @SerializedName("sortBy") val sortBy: String = "name",
    @SerializedName("direction") val direction: String = "ASC"
)